streams/src/main/java/org/apache/kafka/streams/kstream/internals/InternalStreamsBuilder.java

```diff
@@ -111,8 +112,10 @@ public InternalStreamsBuilder(final InternalTopologyBuilder internalTopologyBuil
     public <K, V> KTable<K, V> table(final String topic,
                                      final ConsumedInternal<K, V> consumed,
                                      final MaterializedInternal<K, V, KeyValueStore<Bytes, byte[]>> materialized) {
-        final String sourceName = newProcessorName(KStreamImpl.SOURCE_NAME);
-        final String tableSourceName = newProcessorName(KTableImpl.SOURCE_NAME);
+        final String sourceName = new NamedInternal(consumed.name())
+                .orElseGenerateWithPrefix(this, KStreamImpl.SOURCE_NAME);
+        final String tableSourceName = new NamedInternal(consumed.name())
+                .suffixWithOrElseGet("-table-source", () -> newProcessorName(KTableImpl.SOURCE_NAME));
```

Comment:
appending `KTable` source operators with `-table-source` is not in the KIP, so we'll either need to remove this or update the KIP

Reply:
I think we need to use some suffix because otherwise we would generate two names, ie, end up with a naming conflict -- problem is, that a `KTable` results in two processors and we need a name for each.

Reply:
@bbejeck actually, the `table()` method creates one SourceNode and one ProcessorNode. We need a way to differentiate those two nodes. I should update the KIP to mention this particularity.

Reply:
Ack, I get it now.  Thanks for clarifying.